### PR TITLE
Add the 'app_metro_status' app

### PR DESCRIPTION
**Pull request by magopian**
I would be glad to have a review on this ;)

Here's an example of what the full json could look like when there's a failure:

```
{

    "app_weather": [
        {
            "temperature_now": 11.22,
            "icon_now": "1",
            "humidity_now": 98
        }
    ],
    "app_parcel": [
        {
            "parcel_carrier": "chronopost",
            "arrival": null,
            "info": null,
            "status": null,
            "parcel": "XX123456789FR"
        }
    ]
}
    "app_metro_status": [
        [
            {
                "end_failure": null,
                "line_name": "a"
            }
        ]
    ]
```

Please take note that there's an extra level of nesting here: we could imagine having multiple "metro app" configured for the same "boite", and having multiple failures (in the future, when there will be more than one metro line in Rennes ;)

We might need to think about some naming for the apps... at the moment it's called "Metro Status", but maybe it should be "Metro Status in Rennes" or something? Also we might rather have a filter on the metro line number in the future?

### CODE REVIEW

**Comment by vrialland** [comment]
I was thinking of the same thing for the bikes app as the previous version only worked with VeloStar (Rennes) but other cities provide similar APIs (Velib in Paris for example).
One implementation would be to have a bike app with several providers with the same methods, and you select a particular provider somewhere.
I'll try to make a quick example to discuss it in details.

File: app_metro_status/apps.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from __future__ import unicode_literals
+
+from django.apps import AppConfig
+
+class AppMetroStatusConfig(AppConfig):
+    name = 'app_metro_status'
+    verbose_name = 'App : Metro Status'
```
Comment (vrialland) [inline app_metro_status/apps.py]: Replace with a translatable string?

File: app_metro_status/models.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    def get_app_dictionary(self):
+        if not self.enabled:
+            return
+
+        # we want to update every VALUES_UPDATE_INTERVAL minutes
+        if timezone.now() <= self.last_activity + timedelta(minutes=settings.VALUES_UPDATE_INTERVAL):
+            return maybe_json(self.failures)
+
+        response = requests.get(settings.STAR_API_URL)
```
Comment (vrialland) [inline app_metro_status/models.py]: requests can use a "params" argument (a dict) to build its query string. 

```python
STAR_API_URL = "{}search?dataset=tco-metro-lignes-etat-tr&facet=nomcourt&apikey={}".format(STAR_API_BASE_URL, STAR_API_KEY)
requests.get(settings.STAR_API_URL)
```

can be changed to 

```python
STAR_API_URL = "{}search".format(STAR_API_BASE_URL)

params = {
    'dataset': 'tco-metro-lignes-etat-tr',
    'facet': 'nomcourt',
    'apikey': STAR_API_KEY
}

response = requests.get(STAR_API_URL, params=params)
```

By the way, facet param is not needed in this case (don't know why they put it by default).

**Comment by magopian** [comment]
Wow, awesome review, thanks so much! Will apply the modifications asap

**Comment by magopian** [comment]
Updated the PR, do you want to give it another look @vrialland ?

**Comment by bgaultier** [comment]
Great work guys !
Regarding your concerns about the app dealing zith only one transportation system by app : do you prefer to have one app where you can choose the transport operator *or* **n** apps for **n** transportation systems ?

**Comment by magopian** [comment]
Well, I guess having *n* apps for *n* transportation systems might make it easier to install the needed app. Eg: I can look for "rennes" in the app name, instead of looking for an app for the subway, and then make sure (from its info? from its config?) that it supports Rennes.

On the other hand, it might mean many many apps, where you could instead have a few ones only. And having many different apps doing the same thing (just with a different API in the background) but each in its own different way might lead to a very high incoherence and frustration for the user.

I guess I'll just ask the question right back at you @bgaultier ;)

**Comment by vrialland** [comment]
Here's what I have at this point for Bikes app (only Star provider at this point, both are work in progress)
- https://github.com/vrialland/laboitepro/tree/bikes
- https://github.com/vrialland/laboitepro/tree/bikes_autocomplete_light (with autocomplete integration)

Velib data is very similar to Star's one except fields naming.

The parcel app also provides several providers, but it's true that it might be frustrating for the user to install an app believing it can provide infos for his/her city and then realize it can't.

Maybe there should be some grouping or tagging for example Transportation > Bikes > Rennes (Star) or Transportation > Subway > RATP (Paris)?

**Comment by bgaultier** [comment]
Damned, I just pushed a `app_metro_status` without looking at this thread. Please have a look at #37

**Pull request closed without merging**